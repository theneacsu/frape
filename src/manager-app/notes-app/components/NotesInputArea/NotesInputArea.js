import React from 'react'
import { connect } from 'react-redux'
import '../NotesForm/NotesForm'
import NotesForm from '../NotesForm/NotesForm'
import { startAddNote } from '../../../../actions/notes-app/notes'
import styles from './NotesInputArea.module.css'

const NotesInputArea = props => (
  <div className={styles.div}>
    <NotesForm
      purpose="add"
      titlePlaceholder="ex: Eat healthier"
      bodyPlaceholder="ex: Remove an unhealthy product evey month"
      onSubmit={note => {
        props.startAddNote(note)
        props.history.push('/notes')
      }}
    />
  </div>
)

const mapDispatchToProps = dispatch => ({
  startAddNote: note => dispatch(startAddNote(note))
})

export default connect(
  undefined,
  mapDispatchToProps
)(NotesInputArea)
