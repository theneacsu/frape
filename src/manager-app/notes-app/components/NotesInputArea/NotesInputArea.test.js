import React from 'react'
import { shallow } from 'enzyme'
import { NotesInputArea } from './NotesInputArea'

test('Should render the NotesInputArea component correctly', () => {
  const wrapper = shallow(<NotesInputArea startAddNote={() => {}} />)
  expect(wrapper).toMatchSnapshot()
})

test('Should call startAddNote with the right data', () => {
  const note = {
    title: 'Write snapshot test',
    body: 'It is awesome'
  }
  const history = {
    push: jest.fn()
  }
  const startAddNoteSpy = jest.fn()
  const wrapper = shallow(
    <NotesInputArea startAddNote={startAddNoteSpy} history={history} />
  )
  wrapper.find('NotesForm').prop('onSubmit')(note)
  expect(startAddNoteSpy).toHaveBeenLastCalledWith(note)
  expect(history.push).toHaveBeenCalledWith('/notes')
})
