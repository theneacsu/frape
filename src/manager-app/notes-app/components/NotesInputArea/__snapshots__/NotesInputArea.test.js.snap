// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the NotesInputArea component correctly 1`] = `
<div
  className="div"
>
  <NotesForm
    bodyPlaceholder="ex: Remove an unhealthy product evey month"
    onSubmit={[Function]}
    purpose="add"
    titlePlaceholder="ex: Eat healthier"
  />
</div>
`;
